File: clients/fireback-ios/firebackios/sdk/modules/currency/CurrencyEntity.swift
import Foundation
class CurrencyEntity : Codable, Identifiable {
    var symbol: String? = nil
    var name: String? = nil
    var symbolNative: String? = nil
    var decimalDigits: Int? = nil
    var rounding: Int? = nil
    var code: String? = nil
    var namePlural: String? = nil
}
class CurrencyEntityViewModel: ObservableObject {
    // improve the fields here
    @Published var symbol: String? = nil
    @Published var name: String? = nil
    @Published var symbolNative: String? = nil
    @Published var decimalDigits: Int? = nil
    @Published var rounding: Int? = nil
    @Published var code: String? = nil
    @Published var namePlural: String? = nil
    func getDto() -> CurrencyEntity {
        var dto = CurrencyEntity()
        dto.symbol = self.symbol
        dto.name = self.name
        dto.symbolNative = self.symbolNative
        dto.decimalDigits = self.decimalDigits
        dto.rounding = self.rounding
        dto.code = self.code
        dto.namePlural = self.namePlural
        return dto
    }
}

File: modules/workspaces/codegen/react-native-new/src/modules/fireback/sdk/modules/currency/CurrencyEntity.ts
import {
    BaseDto,
    BaseEntity,
} from "../../core/definitions"
// In this section we have sub entities related to this object
// Class body
export type CurrencyEntityKeys =
  keyof typeof CurrencyEntity.Fields;
export class CurrencyEntity extends BaseEntity {
  public children?: CurrencyEntity[] | null;
  public symbol?: string | null;
  public name?: string | null;
  public symbolNative?: string | null;
  public decimalDigits?: number | null;
  public rounding?: number | null;
  public code?: string | null;
  public namePlural?: string | null;
  public static Navigation = {
      edit(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/currency/edit/${uniqueId}`;
      },
      create(locale?: string) {
          return `${locale ? '/' + locale : ''}/currency/new`;
      },
      single(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/currency/${uniqueId}`;
      },
      query(params: any = {}, locale?: string) {
          return `${locale ? '/' + locale : ''}/currencies`;
      },
      /**
      * Use R series while building router in CRA or nextjs, or react navigation for react Native
      * Might be useful in Angular as well.
      **/
      Redit: "currency/edit/:uniqueId",
      Rcreate: "currency/new",
      Rsingle: "currency/:uniqueId",
      Rquery: "currencies",
  };
  public static definition = {
  "name": "currency",
  "http": {},
  "gormMap": {},
  "fields": [
    {
      "name": "symbol",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "name",
      "type": "string",
      "translate": true,
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "symbolNative",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "decimalDigits",
      "type": "int64",
      "computedType": "number",
      "gormMap": {}
    },
    {
      "name": "rounding",
      "type": "int64",
      "computedType": "number",
      "gormMap": {}
    },
    {
      "name": "code",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "namePlural",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    }
  ],
  "cliShort": "curr",
  "cliDescription": "List of all famous currencies, both internal and user defined ones"
}
public static Fields = {
  ...BaseEntity.Fields,
      symbol: 'symbol',
      name: 'name',
      symbolNative: 'symbolNative',
      decimalDigits: 'decimalDigits',
      rounding: 'rounding',
      code: 'code',
      namePlural: 'namePlural',
}
}